### ISSUE TITLE
Windows coudn't get updates with Adguard connected

### ISSUE THREAD

**Issue by shamarin** [AdguardTeam/AdGuardVPNForWindows#663]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard VPN version

2.2.4

### Operating mode

VPN

### Environment

```markdown
- OS: Windows 11 Pro x64 beta
- Other network-level software: OpenVPN, Kaspersky Plus
```

### Issue Details

Steps to reproduce:
1. Connect to any Adguard VPN service
2. Go to Windows settings and get to system updates
3. Check updates, but Windows will show that there is no updates

### Expected Behavior

Windows will show update or updates and will download them.

### Actual Behavior

In Russia we have blocked system updates from Microsoft via beta or any other channels with russian IP's. Previosly it was ok just before installing Kaspersky Secure protection VPN service (with it while connected to any Kaspersky VPN server Windows is downloading updates as ususal). Internet from all other app is available via Adguard VPN.

### Screenshots

<details><summary>Screenshot 1:</summary>

</details>

**Comment by shamarin** [comment]
Also checked it with the latest beta version of the Adguard VPN, just the same behavior.

**Comment by vankos** [comment]
@shamarin, hi!

Could you clarify a few things for me, please? You mentioned:

> Previosly it was ok just before installing Kaspersky Secure protection VPN service

Do I understand correctly that the problem appears only after installing the Kaspersky VPN? Does the problem persist if you disable Kaspersky VPN? What about if you uninstall it?

You mentioned this issue in [#661](https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661). Is OpenVPN enabled during the problem as well? Is the problem reproduced if you disable both Kaspersky and OpenVPN VPNs?

**Comment by shamarin** [comment]
> @shamarin, hi!
> 
> Could you clarify a few things for me, please? You mentioned:
> 
> > Previosly it was ok just before installing Kaspersky Secure protection VPN service
> 
> Do I understand correctly that the problem appears only after installing the Kaspersky VPN? Does the problem persist if you disable Kaspersky VPN? What about if you uninstall it?
> 
> You mentioned this issue in [#663](https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661). Is OpenVPN enabled during the problem as well? Is the problem reproduced if you disable both Kaspersky and OpenVPN VPNs?

Hi!
1. If to uninstall Kaspersky secure connection when the problem exist also and also were before it sometimes.
2. OpenVPN is not enabled during the problem.
3. The problem persist if to disable both Kaspersky and OpenVPN.

**Comment by vankos** [comment]
I noticed you have Kaspersky antivirus in exclusions, but you have excluded the UI process (avpui.exe), not the main one (avp.exe).

Could you please:

1. Install the latest nightly version [from this link](https://static.adtidy.net/windows/nightly/installer-2.3.1410.0.exe).
2. Manually add the path C:\Program Files (x86)\Kaspersky Lab\Kaspersky 21.17\avp.exe to exclusions (Exclusions -> Apps -> Add application -> Browse).
3. Enable verbose logs.
4. Reproduce the issue.

If the issue persists, please send us the new logs along with the current issue number. You can find the [logs instructions here](https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661#issuecomment-2063395613).

**Comment by shamarin** [comment]
> I noticed you have Kaspersky antivirus in exclusions, but you have excluded the UI process (avpui.exe), not the main one (avp.exe).
> 
> Could you please:
> 
>     1. Install the latest nightly version [from this link](https://static.adtidy.net/windows/nightly/installer-2.3.1410.0.exe).
> 
>     2. Manually add the path C:\Program Files (x86)\Kaspersky Lab\Kaspersky 21.17\avp.exe to exclusions (Exclusions -> Apps -> Add application -> Browse).
> 
>     3. Enable verbose logs.
> 
>     4. Reproduce the issue.
> 
> 
> If the issue persists, please send us the new logs along with the current issue number. You can find the [logs instructions here](https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661#issuecomment-2063395613).

But even without any exclusions Windows couldn't get updates with Adguard connected.

**Comment by shamarin** [comment]
> > I noticed you have Kaspersky antivirus in exclusions, but you have excluded the UI process (avpui.exe), not the main one (avp.exe).
> > Could you please:
> > ```
> > 1. Install the latest nightly version [from this link](https://static.adtidy.net/windows/nightly/installer-2.3.1410.0.exe).
> > 
> > 2. Manually add the path C:\Program Files (x86)\Kaspersky Lab\Kaspersky 21.17\avp.exe to exclusions (Exclusions -> Apps -> Add application -> Browse).
> > 
> > 3. Enable verbose logs.
> > 
> > 4. Reproduce the issue.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If the issue persists, please send us the new logs along with the current issue number. You can find the [logs instructions here](https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661#issuecomment-2063395613).
> 
> But even without any exclusions Windows couldn't get updates with Adguard connected.

allready before this comment tested that nightly build, situation is just the same.

**Comment by shamarin** [comment]
> I noticed you have Kaspersky antivirus in exclusions, but you have excluded the UI process (avpui.exe), not the main one (avp.exe).
> 
> Could you please:
> 
>     1. Install the latest nightly version [from this link](https://static.adtidy.net/windows/nightly/installer-2.3.1410.0.exe).
> 
>     2. Manually add the path C:\Program Files (x86)\Kaspersky Lab\Kaspersky 21.17\avp.exe to exclusions (Exclusions -> Apps -> Add application -> Browse).
> 
>     3. Enable verbose logs.
> 
>     4. Reproduce the issue.
> 
> 
> If the issue persists, please send us the new logs along with the current issue number. You can find the [logs instructions here](https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661#issuecomment-2063395613).

Done, sended today.

**Comment by vankos** [comment]
Thanks for the answers!
Could you please try this test build: 
https://uploads.adguard.org/AdGuardVpn-windowsUpdates.exe

**Comment by shamarin** [comment]
> Thanks for the answers! Could you please try this test build: https://uploads.adguard.org/AdGuardVpn-windowsUpdates.exe

Will try.
What about issue #661???

**Comment by vankos** [comment]
#661 is still under investigation by my colleagues.

**Comment by vankos** [comment]
Hi @shamarin,
Did the build help?

**Comment by shamarin** [comment]
> Hi @shamarin, Did the build help?

Hi! Couldn't test it because now there is no new build on beta channel on Windows 11. So waiting, i think that this week one will be available and i will test this build.

**Comment by shamarin** [comment]
@vankos build 2.3.0 nightly 30 1413 contains this issue fixing as in the build that you gave me in the link?

**Comment by vankos** [comment]
No, it's not. It's an experimental build that has changes not found in any of the official builds yet.

**Comment by shamarin** [comment]
> No, it's not. It's an experimental build that has changes not found in any of the official builds yet.

Ok. Will check it today, because new beta build is available.

**Comment by shamarin** [comment]
> Hi @shamarin, Did the build help?

Hi, @vankos !
Checked today, it works now with this build that you gave me.

**Comment by shamarin** [comment]
Fix of this issue will be included only in 2.4 version?

**Comment by vankos** [comment]
Great news! Thanks for the help! 
Yes, it will be in 2.4, and before that, in one of the nightly builds, probably the next one or the one after that.

**Comment by shamarin** [comment]
> Great news! Thanks for the help! Yes, it will be in 2.4, and before that, in one of the nightly builds, probably the next one or the one after that.

Great! Thank you! Saw that 2.4 will be 64 bit also? That nightly will have 64 bit version also?
